lib/rbeapi/api/routemaps.rb

```diff
@@ -43,63 +43,439 @@ module Api
+      #    hash.
+      #
+      def parse_entries(entries)
+        entries.each_with_object([]) do |rm|
           mdata = /route-map\s(.+)\s(.+)\s(\d+)$/.match(rm)
           rules = get_block(rm)
-          rule_hsh = { 'action' => mdata[2], 'seqno' => mdata[3],
-                       'match_rules' => [], 'set_rules' => [],
-                       'continue_rules' => [] }
-
-          parsed = rules.split("\n").each_with_object({}) do |rule, hsh|
-            mdata = /\s{3}(\w+)\s/.match(rule)
-            case mdata.nil? ? nil : mdata[1]
-            when 'match'
-              hsh['match_rules'] = [] unless hsh.include?('match')
-              hsh['match_rules'] << rule.strip
-            when 'set'
-              hsh['set_rules'] = [] unless hsh.include?('set')
-              hsh['set_rules'] << rule.strip
-            when 'continue'
-              hsh['continue_rules'] = [] unless hsh.include?('continue')
-              hsh['continue_rules'] << rule.strip
+          rule_hsh = { action: mdata[2], seqno: mdata[3].to_i }
+          unless rules.nil?
+            rules.split("\n").each_with_object({}) do |rule|
+              parse_rule(rule, rule_hsh)
             end
           end
-          rule_hsh.update(parsed)
-          arry << rule_hsh
+          return rule_hsh
         end
       end
+      private :parse_entries
 
-      def getall
-        maps = config.scan(/(?<=^route-map\s)[^\s]+/)
-        maps.each_with_object({}) do |name, hsh|
-          hsh[name] = get name unless hsh.include?(name)
+      ##
+      # parse rule is a private method to parse a rule.
+      #
+      # @return [Hash<Symbol, Object>] returns a hash that represents the
+      #   rules for routemaps from the nodes running configuration.  If
+      #   there are no routemaps configured, this method will return an empty
+      #    hash.
+      #
+      def parse_rule(rule, rule_hsh)
+        mdata = /\s{3}(\w+)\s/.match(rule)
+        case mdata.nil? ? nil : mdata[1]
+        when 'match'
+          rule_hsh[:match_rules] = [] unless rule_hsh.include?(:match_rules)
+          rule_hsh[:match_rules] << rule.sub('match', '').strip
+        when 'set'
+          rule_hsh[:set_rules] = [] unless rule_hsh.include?(:set_rules)
+          rule_hsh[:set_rules] << rule.sub('set', '').strip
+        when 'continue'
+          rule_hsh[:continue] = nil unless rule_hsh.include?(:continue)
+          rule_hsh[:continue] = rule.sub('continue', '').strip.to_i
+        when 'description'
+          rule_hsh[:description] = nil unless rule_hsh.include?(:description)
+          rule_hsh[:description] = rule.sub('description', '').strip
+        end
+      end
+      private :parse_rule
+
+      ##
+      # name_commands is utilized by create to prepare the specified
+      # routemap.
+      #
+      def name_commands(name, opts)
+        if opts[:enable] == false
+          @cmd = "no route-map #{name}"
```

Comment:
I used a global to pass @cmd from name_commands to parse_entries. I could also just return cmd without using a variable if you prefer.